### ISSUE TITLE
THEMIS template creation reads a concrete qhac max inside the traced region: exact-path predict_photometry gradient dies with ConcretizationTypeError

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2114]
## Symptom

`jax.grad` of exact-path `predict_photometry` (build with `approx=None`) on a model containing THEMIS dust emission dies with `ConcretizationTypeError`. Found by the #2092 benchmark-honesty work: the kitchen-sink gradient sections (`nebular_ssp + themis + AGN + radio + xray`, all three SFH types) have never been measurable — the harness printed only the exception class name, so the location was unknown until now.

## Mechanism (measured, traceback in hand)

The deepest tengri frame is `src/tengri/components/dust/emission_templates.py:2343`, in `_qhac_axis_to_cigale`:

```python
if float(jnp.max(qhac_grid)) > 0.5:
```

The function's own docstring states the contract honestly — "**JIT-compatible**: no — reads a concrete max to pick a convention" — and the call site at `emission_templates.py:2393-2401` (inside `create_themis_from_grid`) carries a comment acknowledging the concrete read. The defect is not the convention check itself but that it is reachable **inside a traced region**: under `jax.grad`, THEMIS template creation runs lazily with `qhac_grid` a tracer, and the `float()` raises.

The load-time call in `components/dust/emission/templates/themis.py:95` is fine (concrete numpy axis). The in-trace path is the bug.

## Blast radius

- Fit surfaces default to `WavePrecomp` since 2026-08-10, so ordinary fits mostly do not take the exact path; custom gradient code, forward-model research code, and the benchmark's gradient sections do. (The precomp-path gradient for the same config is untested — the #2092 harness returns after the exact-path failure — so whether `WavePrecomp` avoids this frame is unknown, not established.)
- Reproducer: `bench/scripts/kitchen_sink_gradient_reproducer.py` on the `fix/2092-bench-honesty` branch (PR forthcoming); it differentiates the free-parameter subdict only, so this is not an artifact of grading Fixed leaves.

## Fix shape (not done here)

Resolve the qhac convention at load/build time where the axis is concrete — e.g. resolve in `load`/template-dict construction and store the converted axis (as `themis.py:95` already does on its path), so `_qhac_axis_to_cigale` is never entered with a tracer; keep the function's loud concrete-read contract for its legitimate concrete callers. The kitchen-sink gradient sections in `bench/scripts/benchmark_forward_model.py` join `_REQUIRED_SECTIONS` when this is fixed (they are excluded with a comment citing this issue).

**Comment by suchethac** [comment]
## Correction to the mechanism (measured — full traceback now in hand)

The original text said the `themis.py:95` load-time call was "fine (concrete numpy axis)" and the in-trace path was elsewhere. That was wrong: **`themis.py:95` is the failing frame.** The verbatim chain from the committed reproducer (`bench/scripts/kitchen_sink_gradient_reproducer.py`, branch `fix/2092-bench-honesty`):

```
predict_photometry (sed_model.py:4816)
→ predict_observables_jit (:7561)
→ _get_or_build_predict_observables_jit (:7617)   # first call
→ _build_component_chain (:8572)
→ comp.precompute (sed_model_component.py:630)
→ self.load(wave_grid) (dust/emission/templates/themis.py:95)
→ _qhac_axis_to_cigale (emission_templates.py:2343)
   if float(jnp.max(qhac_grid)) > 0.5:   # ConcretizationTypeError
```

The axis IS concrete HDF5 data — the defect is *where* the code runs: the component chain is built **lazily inside the first traced call**, so `load()` executes under `jax.grad`/`jax.jit` tracing, where `jnp.max` on a concrete array returns a lifted tracer constant and `float()` raises.

Two independent fix shapes (either suffices; both are defensible together):
1. `_qhac_axis_to_cigale` uses `float(np.max(np.asarray(qhac_grid)))` — the axis is grid data from disk, never legitimately traced, and numpy keeps the read concrete under any trace. The function's "reads a concrete max" contract stays true and becomes trace-immune.
2. Hoist `_build_component_chain` (or at least each component's `precompute/load`) out of the traced region — a first call under an outer `jax.grad` currently does template I/O and convention resolution inside the trace, which is also a compile-cost and #1738-adjacent state concern beyond this one crash.

**Comment by suchethac** [comment]
Claimed (orchestrator session b5ec05, 2026-09-05): implementing the concrete-read fix for the qhac convention probe so the exact-path gradient no longer dies under trace, per the corrected mechanism in the comment above. PR to follow.
